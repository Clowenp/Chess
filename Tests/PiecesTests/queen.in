q 0 0 6 0
q 4 3 6 4
q 4 4 4 7
q 7 4 8 4
q 4 0 7 3
