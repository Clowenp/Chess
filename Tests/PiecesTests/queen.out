i am a queen at [0,0]
my possible paths are: 
< [0,1] [0,2] [0,3] [0,4] [0,5] [0,6] [0,7] >
< [1,1] [2,2] [3,3] [4,4] [5,5] [6,6] [7,7] >
< [1,0] [2,0] [3,0] [4,0] [5,0] [6,0] [7,0] >
< >
< >
< >
< >
< >
the move to [6,0] is possible
i am a queen at [4,3]
my possible paths are: 
< [4,4] [4,5] [4,6] [4,7] >
< [5,4] [6,5] [7,6] >
< [5,3] [6,3] [7,3] >
< [5,2] [6,1] [7,0] >
< [4,2] [4,1] [4,0] >
< [3,2] [2,1] [1,0] >
< [3,3] [2,3] [1,3] [0,3] >
< [3,3] [2,3] [1,3] [0,3] >
the move to [6,4] is NOT possible
i am a queen at [4,4]
my possible paths are: 
< [4,5] [4,6] [4,7] >
< [5,5] [6,6] [7,7] >
< [5,4] [6,4] [7,4] >
< [5,3] [6,2] [7,1] >
< [4,3] [4,2] [4,1] [4,0] >
< [3,3] [2,2] [1,1] [0,0] >
< [3,4] [2,4] [1,4] [0,4] >
< [3,4] [2,4] [1,4] [0,4] >
the move to [4,7] is possible
i am a queen at [7,4]
my possible paths are: 
< [7,5] [7,6] [7,7] >
< >
< >
< >
< [7,3] [7,2] [7,1] [7,0] >
< [6,3] [5,2] [4,1] [3,0] >
< [6,4] [5,4] [4,4] [3,4] [2,4] [1,4] [0,4] >
< [6,4] [5,4] [4,4] [3,4] [2,4] [1,4] [0,4] >
the move to [8,4] is NOT possible
i am a queen at [4,0]
my possible paths are: 
< [4,1] [4,2] [4,3] [4,4] [4,5] [4,6] [4,7] >
< [5,1] [6,2] [7,3] >
< [5,0] [6,0] [7,0] >
< >
< >
< >
< [3,0] [2,0] [1,0] [0,0] >
< [3,0] [2,0] [1,0] [0,0] >
the move to [7,3] is possible
